lib/tesla/adapter/gun.ex

```diff
@@ -148,6 +148,20 @@ if Code.ensure_loaded?(:gun) do
           opts
         end
 
+      # Fix for FollowRedirects: SERVER ALERT: Fatal - Handshake Failure
+      opts =
+        if uri.scheme == "https" do
+          host = uri.host |> to_charlist()
+
+          tls_opts =
+            Map.get(opts, :tls_opts, %{})
+            |> Map.put(:server_name_indication, host)
+
+          Map.put(opts, :tls_opts, tls_opts)
+        else
+          opts
+        end
+
```

Comment:
Any chances for test for this case?
Also, what does SNI has to do with FollowRedirects middleware? 🤔

Reply:
I think nope, because server_name_indication option is transferred to :ssl.connect and is not returned to the %Env{} in response. Gun opens tls connection through ip, that's why default sni option with host is not added by ssl module (http://erlang.org/doc/man/ssl.html#type-sni). SNI option is needed for some certificates, on which gun returns Handshake error. This option is tested on dev server :)

Reply:
Is it needed test for this [case](https://github.com/teamon/tesla/pull/310/files#diff-f93babd4700a230f6ced434c9dd3a68eR175)?